gh-137836: Support more RAWTEXT and PLAINTEXT elements in HTMLParser (#137837)
python/cpython: Doc/library/html.parser.rst

This module defines a class :class:`HTMLParser` which serves as the basis for
parsing text files formatted in HTML (HyperText Mark-up Language) and XHTML.

.. class:: HTMLParser(*, convert_charrefs=True, scripting=False)

   Create a parser instance able to parse invalid markup.

   If *convert_charrefs* is ``True`` (the default), all character

>    references (except the ones in ``script``/``style`` elements) are

Reviewer: This should be updated now that the list has been expanded.

It might be easier to have a short section about parsing modes, listing each mode, which elements trigger it, whether charrefs are converted or not, and when the state is terminated.

Here we could then say
```suggestion
   references (except the ones in RAWTEXT tags) are
```
with `RAWTEXT` linking to that section.

Author: Do we need to document this here? This is a part of the HTML5 specification. What will the user get from this information?

   automatically converted to the corresponding Unicode characters.

   If *scripting* is true, the ``noscript`` element is parsed in the

>    RAWTEXT mode.

Reviewer: ```suggestion
   If *scripting* is false (the default), the content of the ``noscript`` element
   is parsed normally; if it's true, it's parsed in RAWTEXT mode.
```
And this could link to the aforementioned section.

It would also be useful to expand a bit on why this exists, either by mentioning briefly that `<noscript>` is used as a fallback when JS is not enabled, or by linking to a page (w3c or mdn) that explains that.

This module defines a class :class:`HTMLParser` which serves as the basis for
parsing text files formatted in HTML (HyperText Mark-up Language) and XHTML.

.. class:: HTMLParser(*, convert_charrefs=True, scripting=False)

   Create a parser instance able to parse invalid markup.

   If *convert_charrefs* is true (the default), all character
   references (except the ones in elements like ``script`` and ``style``) are
   automatically converted to the corresponding Unicode characters.

   If *scripting* is false (the default), the content of the ``noscript``
   element is parsed normally; if it's true, it's parsed in RAWTEXT mode,

>    like ``script``.

Reviewer: ```suggestion
   If *scripting* is false (the default), the content of the ``noscript``
   element is parsed normally; if it's true, it's returned as is without
   being parsed.
```

   .. versionchanged:: 3.5
      The default value for argument *convert_charrefs* is now ``True``.

>    .. versionchanged:: 3.13.8

Author: ```suggestion
   .. versionchanged:: 3.14.1
```

>       Added the *scripting* parameter.

Reviewer: This should probably mention the other changes too.

Author: Other changes are considered bugfixes (even security fixes). They will be backported up to 3.9.

We only document the `scripting` parameter because it is a part of the public API.
